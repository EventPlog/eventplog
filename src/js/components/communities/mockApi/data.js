export default {
  communities: [
    {
      id: 1,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing.',
      featured_image: '/login-bg.jpg',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: true,
    },
    {
      id: 2,
      name: 'Google Developer Group',
      description: 'Developer community bringing up community interest, powereed by google.',
      featured_image: '/tech_is_in_you.png',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Developers',
      joined: true,
    },
    {
      id: 3,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing',
      featured_image: '/login-bg.jpg',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: true,
    },
    {
      id: 4,
      name: 'Google Developer Group',
      description: 'Developer community bringing up community interest, powereed by google.',
      featured_image: null,
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Developers',
      joined: false,
    },
    {
      id: 5,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing',
      featured_image: '/tech_is_in_you.png',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: false,
    },
    {
      id: 6,
      name: 'Google Developer Group',
      description: 'Developer community bringing up community interest, powereed by google.',
      featured_image: null,
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Developers',
      joined: false,
    }
  ],

  events: [
    {
      id: 1,
      title: 'Tech is in you',
      featured_image: '/tech_is_in_you.png',
      interested_persons: 211,
      interested: false,
      community: {
        id: 1,
        name: 'Google Developer Group',
        description: 'Developer community bringing up community interest, powereed by google.',
        featured_image: '/login-bg.jpg',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Developers',
        joined: false,
      }
    },
    {
      id: 2,
      title: 'Writers corner workshop',
      featured_image: '/login-bg.jpg',
      interested_persons: 128,
      interested: false,
      community: {
        id: 2,
        name: 'Facebook Developer Circle Lagos',
        description: 'A community of developers united in knowledge sharing',
        featured_image: '/tech_is_in_you.png',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Technology',
        joined: false,
      }
    },
    {
      id: 3,
      title: 'Tech is in you',
      featured_image: '/tech_is_in_you.png',
      interested_persons: 221,
      interested: true,
      community: {
        id: 1,
        name: 'Google Developer Group',
        description: 'Developer community bringing up community interest, powereed by google.',
        featured_image: '/login-bg.jpg',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Developers',
        joined: false,
      }
    },
    {
      id: 4,
      title: 'Writers corner workshop',
      featured_image: '/login-bg.jpg',
      interested_persons: 18,
      interested: true,
      community: {
        id: 2,
        name: 'Facebook Developer Circle Lagos',
        description: 'A community of developers united in knowledge sharing',
        featured_image: '/tech_is_in_you.png',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Technology',
        joined: false,
      }
    }
  ]
}