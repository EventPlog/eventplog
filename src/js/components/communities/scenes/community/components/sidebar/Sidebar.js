import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './sidebar.scss';
import styled, { ThemeProvider } from 'styled-components';
import defaults from 'js/styles/theme/variables';

const Aside = styled.aside`
  width: 200px;
  border-right: 1px solid #eee;
  height: 100vh;

  > ul {
    margin: 100px 30px 0 0px;
    text-align: right;
    line-height: 60px;
    list-style: none;

    .active {
      border-bottom: 1px solid ${defaults.activeLink};
      border-bottom: 1px solid var(--activeLink, ${defaults.activeLink});
      margin-bottom: 15px;
      padding: 0 0 15px 50px;
    }
    
    a {
      color: ${props => props.theme.fg};
    }
    
    > li {
    
    }
  }
`

Aside.defaultProps = {
}

const Sidebar = ({ themeColors, event = {} }) => {
  const menuItems = ["Tasks", "Check-in", "Guests", "Settings"];
  return (
      <Aside>
        <ul>
          {
            menuItems.map((menuItem, index) =>
              <li key={index}>
                <NavLink to={`/events/${event.id}/${menuItem.toLowerCase()}`}
                         activeClassName="active">
                  {menuItem}
                </NavLink>
              </li>
            )
          }
        </ul>
      </Aside>
  )

}

export default Sidebar;
