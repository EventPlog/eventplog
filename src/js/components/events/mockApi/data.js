export default {
  communities: [
    {
      id: 1,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing.',
      featured_image: '/login-bg.jpg',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: true,
    },
    {
      id: 2,
      name: 'Google Developer Group',
      description: 'Developer community bringing up community interest, powereed by google.',
      featured_image: '/tech_is_in_you.png',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Developers',
      joined: true,
    },
    {
      id: 3,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing',
      featured_image: '/login-bg.jpg',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: true,
    },
  ],

  events: [
    {
      id: 1,
      title: 'Tech is in you',
      featured_image: '/tech_is_in_you.png',
      interested: true,
      interested_persons: 211,
      start_date: '28 June, 2018',
      start_time: '10:00am',
      end_date: '28 June, 2018',
      end_time: '01:00pm',
      venue: 'StartZone, Gbagada',
      community: {
        id: 3,
        name: 'Google Developer Group',
        description: 'Developer community bringing up community interest, powereed by google.',
        featured_image: '/login-bg.jpg',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Developers',
        joined: false,
      }
    },
    {
      id: 2,
      title: 'Writers corner workshop',
      featured_image: '/login-bg.jpg',
      interested: true,
      interested_persons: 211,
      start_date: '28 June, 2018',
      start_time: '10:00am',
      end_date: '28 June, 2018',
      end_time: '01:00pm',
      venue: 'StartZone, Gbagada',
      community: {
        id: 1,
        name: 'GDG Lagos',
        description: 'A community of developers united in knowledge sharing',
        featured_image: '/tech_is_in_you.png',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Technology',
        joined: false,
      }
    },
    {
      id: 3,
      title: 'Figma Design Series',
      featured_image: '/tech_is_in_you.png',
      interested: false,
      interested_persons: 108,
      start_date: '28 June, 2018',
      start_time: '10:00am',
      end_date: '28 June, 2018',
      end_time: '01:00pm',
      venue: 'StartZone, Gbagada',
      community: {
        id: 2,
        name: 'Facebook Developer Circle Lagos',
        description: 'A community of developers united in knowledge sharing',
        featured_image: '/tech_is_in_you.png',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Technology',
        joined: false,
      }
    }
  ]
}