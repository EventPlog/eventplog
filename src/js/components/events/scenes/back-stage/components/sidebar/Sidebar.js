import React from 'react';
import { NavLink } from 'react-router-dom';
import styled, { ThemeProvider } from 'styled-components';
import { Icon } from 'semantic-ui-react'

import defaults from 'js/styles/theme/variables';
import { media, maxMedia } from 'js/styles/mixins'

const Aside = styled.aside`
  width: 200px;
  border-right: 1px solid #eee;
  height: 100vh;
  background: #eee;

  ${
    media.tablet`
      width: 100px;
    `
  }
  
  ${
    media.phone`
      height: 50px;
    `
  }
  
  > ul {
    margin: 100px 30px 0 0px;
    text-align: right;
    line-height: 60px;
    list-style: none;

    .active {
      border-bottom: 1px solid ${defaults.activeLink};
      border-bottom: 1px solid var(--activeLink, ${defaults.activeLink});
      margin-bottom: 15px;
      padding: 0 0 15px 50px;
      color: #444;
      
      ${
        media.phone`
          padding: 0.5rem;
          border-bottom: none;
        `
      }
    }
    
    
    ${
      media.phone`
        display: flex;
        font-size: 90%;
        margin: 0 2rem;
        justify-content: space-between;
        line-height: 50px;
      `
    }
    
    i {
      margin-right: 1rem;
    }
  }
`

Aside.defaultProps = {
}

const Sidebar = ({ className, event = {} }) => {
  const menuItems = [
    { name: "Tasks", icon: 'settings' },
    { name: "Guests", icon: 'users' },
    { name: "Feedback", icon: 'send' },
    {name: "Settings", icon: 'settings' }
  ];
  return (
    <ThemeProvider theme={{
      ...defaults,
    }}>
      <Aside className={className}>
        <ul>
          {
            menuItems.map(({name, icon}, index) =>
              <li key={index}>
                <NavLink to={`/communities/1/events/${1}/backstage/${name.toLowerCase()}`}
                         activeClassName="active">
                  <span className="hidden-xs">
                    <Icon name={icon || 'users'} />
                  </span>
                  {name}
                </NavLink>
              </li>
            )
          }
        </ul>
      </Aside>
    </ThemeProvider>
  )

}

export default Sidebar;
