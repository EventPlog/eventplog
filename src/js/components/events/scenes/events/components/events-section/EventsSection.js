import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

// internal
import ContentPanel from 'js/components/shared/content-panel'
import Loading from 'js/components/shared/loading'
import Error from 'js/components/shared/loading/Error'
import Button from 'js/components/shared/button'
import Pagination from 'js/components/shared/pagination'
import { pluralize } from 'js/utils'

export const generateTitle = (event = {}) => {
  const community = event.community || {}
  return (
    <Link to={`/communities/${event.community_id}/events/${event.id}`}>
      {event.title}
    </Link>
  )
}

export const generateDescription = (community = {}) => (
  <span>
    By <Link to={`/communities/${community.id}`}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event = {}) => ([
  <ul key={`date${event.id}`}>
    <li>
      {event.date}
    </li>
    <li>
      {event.time}
    </li>
    <li>
      {event.venue}
    </li>
  </ul>,
  <ul key={`interest${event.id}`}>
    <li>
      {event.interested_persons} {pluralize('person', event.interested_persons)} interested
    </li>
  </ul>
])

export const generateCTA = (handleClick) => (
  <Button onClick={handleClick}>
    Interested
  </Button>
)

const EventsSection = ({
  title,
  events = {},
  attendEvent,
  getEvents,
}) => {
  const {loading, error, data = [], meta = {}} = events;
  const shouldDisplayData = (!loading && !error && data);
  return (
    <ContentPanel title={title}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {shouldDisplayData && data.map(({featured_image, ...event}) => {
          const title = generateTitle(event)
          const description = generateDescription(event.community)
          const meta = generateMeta(event)
          const btn = {onClick: () => attendEvent(event), text: 'interested'}
          const titleLink = `/communities/${event.community_id}/events/${event.id}`

          return (
            <ContentPanel.Card
              key={event.id}
              {...{title, description, featured_image, meta, btn, titleLink}}
              showButton={!event.is_attending} />
          )
        }
      )}
      {shouldDisplayData && data.length < 1 && <p>No events to display today ...</p>}
      {
        meta && meta.total_pages && (data.length > 0 || meta.current_page > 1)
          ? <Pagination totalPages={meta.total_pages}
                        activePage={meta.current_page}
                        onPageChange={getEvents} />
          : ''
      }
    </ContentPanel>
  )
}

export default EventsSection