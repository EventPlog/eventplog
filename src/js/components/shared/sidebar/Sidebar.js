import React from 'react'
import styled from 'styled-components'
import { media } from 'js/styles/mixins'
import SidebarCard from './SidebarCard'
import { lighten } from 'polished'

import SidebarEvents from './events-section'
import SidebarCommunities from './communities-section'
import colors from 'js/styles/theme/variables'

const StyledSidebar = styled.div`
  color: ${props => lighten(0.25, props.theme.activeLink)};
  border-radius: 10px;
  background: ${props => props.theme.activeLinkBg};
  
  p {
    color: ${lighten(0.4, colors.blue)};
  }
  
  ${
    media.phone`
      border-radius: 0;
    `
  }
  
  .header {
    border-bottom: 1px solid #ccc;
    margin: 1rem 0;
    font-size: 1.2rem;
    color: ${props => lighten(0.4, props.theme.activeLink)};
    font-weight: 500;
  }
`

const Sidebar = function({
  title,
  children,
  ...otherProps
}) {
  return (
    <StyledSidebar className="sidebar" {...otherProps}>
      <h5 className="header">
        {title}
      </h5>
      {children}
    </StyledSidebar>
  )
}

Sidebar.Card = SidebarCard
Sidebar.Events = SidebarEvents
Sidebar.Communities = SidebarCommunities

export default Sidebar