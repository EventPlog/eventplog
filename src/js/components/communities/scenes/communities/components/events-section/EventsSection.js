import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'
import { genEventLink, genCommunityLink } from 'js/utils'

export const generateTitle = (event = {}, community = {}) => (
  <Link to={genEventLink(event, community)}>
    {event.title}
  </Link>
)

export const generateDescription = (community) => (
  <span>
    By <Link to={genCommunityLink(community)}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} people interested`
)


const EventsSection = ({ events }) => {
  const {loading, error, data } = events
  return (
    <Sidebar title="Events you may like">
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event, community);
          const description = community ? generateDescription(community) : '';
          const meta = generateMeta(event)
          return <Sidebar.Card
                    key={event.id}
                    {...{title, description, featured_image, meta}} />
        }
      )}
    </Sidebar>
  )
}

export default EventsSection