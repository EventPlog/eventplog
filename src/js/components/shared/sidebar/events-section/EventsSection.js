import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'

export const generateTitle = (event = {}) => (
  <Link to={`/communities/${event.community_id}/events/${event.id}`}>
    {event.title}
  </Link>
)

export const generateDescription = (community) => (
  <span>
    By <Link to={`/communities/${community.id}`}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} people interested`
)


const EventsSection = ({
  title,
  events = {data: [], meta: {}},
  attendEvent
}) => {
  const {loading, error, data = {} } = events
  return (
    <Sidebar title={title || "Events you may like"}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event);
          const description = generateDescription(community);
          const meta = generateMeta(event)
          const btn = event.is_attending
                      ? {}
                      : {onClick: () => attendEvent(event), text: 'Interested'}
          return (
            <Link to={`/communities/${event.community_id}/events/${event.id}`}>
              <Sidebar.Card key={event.id}
                             {...{title, description,
                             featured_image, meta, btn}} />
            </Link>
          )
        }
      )}
    </Sidebar>
  )
}

export default EventsSection