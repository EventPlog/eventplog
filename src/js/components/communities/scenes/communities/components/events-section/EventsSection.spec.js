
import React from 'react';
import EventsSection, {
  generateTitle,
  generateDescription,
  generateMeta
} from './EventsSection';
import { shallow } from 'enzyme';

import Sidebar from 'js/components/shared/sidebar'
import data from 'js/mock-api/data'

describe('EventsSection', () => {
  const events = data.events

  it('should render correctly', () => {
    const wrapper = shallow( <EventsSection events={events} /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(Sidebar).length).toEqual(1);
    expect(wrapper.find(Sidebar.Card).length).toEqual(events.length);
  })

  it('should pass the right props to children', () => {
    const wrapper = shallow( <EventsSection events={events} />)

    const sidebarCardInstances = wrapper.find(Sidebar.Card)

    expect(sidebarCardInstances.at(0).props().title).toEqual(generateTitle(events[0], events[0].community.id))
    expect(sidebarCardInstances.at(0).props().description).toEqual(generateDescription(events[0].community))
    expect(sidebarCardInstances.at(0).props().featured_image).toEqual(events[0].featured_image)
    expect(sidebarCardInstances.at(0).props().meta).toEqual(generateMeta(events[0]))
  })
});