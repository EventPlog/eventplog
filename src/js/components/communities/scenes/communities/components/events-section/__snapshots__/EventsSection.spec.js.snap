// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Communities::Communities::EventsSection should render correctly 1`] = `
<Sidebar
  title="Events you may like"
>
  <SidebarCard
    description={
      <span>
        By 
        <Link
          replace={false}
          to="/c/1"
        >
          Facebook Developer Circle Lagos
        </Link>
      </span>
    }
    featured_image="/tech_is_in_you.png"
    key="1"
    meta="211 people interested"
    title={
      <Link
        replace={false}
        to="/c/1/e/1"
      >
        Tech is in you
      </Link>
    }
  />
  <SidebarCard
    description={
      <span>
        By 
        <Link
          replace={false}
          to="/c/2"
        >
          Google Developer Group
        </Link>
      </span>
    }
    featured_image="/login-bg.jpg"
    key="2"
    meta="211 people interested"
    title={
      <Link
        replace={false}
        to="/c/2/e/2"
      >
        Writers corner workshop with writing and use of English
      </Link>
    }
  />
  <SidebarCard
    description={
      <span>
        By 
        <Link
          replace={false}
          to="/c/3"
        >
          ForLoop Lagos
        </Link>
      </span>
    }
    featured_image="/login-bg.jpg"
    key="3"
    meta="108 people interested"
    title={
      <Link
        replace={false}
        to="/c/3/e/3"
      >
        Figma Design Series
      </Link>
    }
  />
</Sidebar>
`;
